docs/meshstack.user-group-provisioning.md

```diff
@@ -0,0 +1,112 @@
+---
+
+Provisioning users and groups to meshStack works based on the [SCIM 2.0](http://www.simplecloud.info/) standard.
+We provide an SCIM API that enables external systems such as an Azure Active Directory (AAD) to directly
+manage up-to-date user and group information within meshStack. `Provisioning` means changes about users and groups
+in the external system are automatically mirrored to the provisioned equivalents in meshStack. For example, when
+the members of a certain AAD group are changing and provisioning is set up, those changes will be reflected in the group within meshStack as well
+without additional effort. That way we can model the complete lifecycle of users and groups automatically.
+
+## Provisioning with SCIM
+
+SCIM clients need to address users and groups in order to manage the creation, updates and deletion. Because they cannot know
+user or group identifiers in advance they will attempt to find a user or group by a unique attribute:
+for users it selects the `userName`, for groups it uses the `displayName`. As a result these fields need to be unique.
+To support a wider range of userName formats, meshStack will look up users by their username and uses the email
+as a fallback.
+
+### Example
+
+The AAD as an SCIM client looks up an existing user by calling:
+
+```sh
+GET .../api/scim/v2/Users?filter=userName+eq+"johndoe@company.com"
+```
+
+Let's assume there is a user with the username `johndoe@company.com`, then meshStack will then answer with an SCIM ListResponse containing
+the matching UserResource, e.g. like this:
+
+```json
+{
+  ...
+  "Resources": [
+    {
+      "schemas": [
+        "urn:ietf:params:scim:schemas:core:2.0:User"
+      ],
+      "id": "54fbbbaa-c80c-49df-b13d-27133cb81740",
+      "userName": "johndoe@company.com",
+      ...
+    }
+  ],
+  ...
+}
+```
+
+From that response the AAD retrieves the user's id and can from now on refer to this user with the identifier `54fbbbaa-c80c-49df-b13d-27133cb81740`.
+Let's further assume that the AAD wants to update the members of a group with the identifier `d90c1657-63b1-44be-8cdc-e3a8ffc5a7d1` and the aforementioned user
+is the only member, then the AAD will send the following PatchOperation:
+
+```sh
+PATCH .../api/scim/v2/Groups/d90c1657-63b1-44be-8cdc-e3a8ffc5a7d1
+{
+    "schemas": [
+        "urn:ietf:params:scim:api:messages:2.0:PatchOp"
+    ],
+    "Operations": [
+        {
+            "op": "Add",
+            "path": "members",
+            "value": [
+                {
+                    "value": "54fbbbaa-c80c-49df-b13d-27133cb81740"
+                }
+            ]
+        }
+    ]
+}
+```
+
+## SCIM API limitations
+
+The SCIM standard introduces comprehensive functionality to work with user and group resources.
+The implementation of meshStack's SCIM API does not support all features of SCIM. Here is a list of current limitations:
+
+- Deletion of users is not supported.
+- Filtering options are reduced:
```

Comment:
this has nothing to do with the scoping of the sync right? This is how the SCIM client can make changes to existing identities in meshStack?

Reply:
clarified that, those are filter options and parameters that you can apply to search users and groups.